adapters/impactify/impactify.go

```diff
@@ -0,0 +1,186 @@
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	if bidRequest.Device != nil {
+		if bidRequest.Device.UA != "" {
+			headers.Add("User-Agent", bidRequest.Device.UA)
+		}
+		// Add IPv4 or IPv6 if available
+		if bidRequest.Device.IP != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IP)
+		} else if bidRequest.Device.IPv6 != "" {
+			headers.Add("X-Forwarded-For", bidRequest.Device.IPv6)
+		}
+	}
+	if bidRequest.Site != nil {
+		headers.Add("Referer", bidRequest.Site.Page)
+	}
+
+	// set user's cookie
+	if bidRequest.User != nil && bidRequest.User.BuyerUID != "" {
+		headers.Add("Cookie", "uids="+bidRequest.User.BuyerUID)
+	}
+
+	adapterRequests = append(adapterRequests, &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+	})
+
+	return adapterRequests, errs
+}
+
+func (a *adapter) MakeBids(
+	internalRequest *openrtb2.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	} else if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Invalid request.",
+		}}
+	} else if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected HTTP status %d.", response.StatusCode),
+		}}
+	}
+
+	var openRtbBidResponse openrtb2.BidResponse
+
+	if err := json.Unmarshal(response.Body, &openRtbBidResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "bad server body response",
+		}}
+	}
+
+	if len(openRtbBidResponse.SeatBid) == 0 {
+		return nil, nil
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(openRtbBidResponse.SeatBid[0].Bid))
+	bidResponse.Currency = openRtbBidResponse.Cur
+
+	sb := openRtbBidResponse.SeatBid[0]
+	for i := 0; i < len(sb.Bid); i++ {
+		if !(sb.Bid[i].Price > 0) {
+			continue
+		}
+		impMediaType, err := getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		bid := sb.Bid[i]
+		bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+			Bid:     &bid,
+			BidType: impMediaType,
+		})
+	}
+	return bidResponse, nil
+}
+
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	var mediaType openrtb_ext.BidType = ""
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Banner != nil {
+				mediaType = openrtb_ext.BidTypeBanner
+				return mediaType, nil
+			} else if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+				return mediaType, nil
+			} else if imp.Native != nil {
+				mediaType = openrtb_ext.BidTypeNative
+				return mediaType, nil
+			}
```

Comment:
You might want to remove the Native option. According to the bidder yaml, this adapter does not support native bids.

Reply:
Done.